### ISSUE TITLE
biobert predicts only one class on test set

### ISSUE THREAD

**Issue by vijay-srivas** [dmis-lab/biobert#44]
Hi.
I am performing a binary text classification task using 'run_classifier.py'. When I look at the predicted probabilities for class 0 and 1 (in 'test_results.tsv'), i see that the predicted probabilities are very similar for all data points and biased towards class 0. The output looks like this-

0.9994677	0.00053235615
0.9994677	0.00053235615
0.9994677	0.00053235615
0.9994677	0.00053235615
0.9994677	0.00053235615
0.9994677	0.00053235615
0.9994677	0.00053235615

This is my config file-
python run_classifier.py 
--task_name=cola 
--do_train=true 
--do_eval=true 
--do_predict=true 
--data_dir=./data_cla_abs/ 
--vocab_file=./pretrained_weights_BIOBERT_DIR/biobert_v1.0_pubmed_pmc/biobert_v1.0_pubmed_pmc/vocab.txt 
--bert_config_file=./pretrained_weights_BIOBERT_DIR/biobert_v1.0_pubmed_pmc/biobert_v1.0_pubmed_pmc/bert_config.json --init_checkpoint=./pretrained_weights_BIOBERT_DIR/biobert_v1.0_pubmed_pmc/biobert_v1.0_pubmed_pmc/biobert_model.ckpt 
--max_seq_length=512 
--train_batch_size=2 
--learning_rate=2e-5 
--num_train_epochs=5.0 
--output_dir=./bert_output_cla_abs/ 
--do_lower_case=False

The data is pubmed abstacts. I am trying to classify pubmed abstracts into into 2 classes. 

Happy to share more info.

Kindly help me understand the cause of this problem.

Thanks.

**Comment by vijay-srivas** [comment]
I suspect the issue could be with class imbalance. Instances with negative class far outnumbers instances with positive class. Can someone also suggest where I can specify class weight to be balanced in (bio)bert?

Thanks

**Comment by oakkas** [comment]
Hi @vijay-srivas,
I am wondering how did you prepare your input data file? Is it a csv file with one column of sentence and second column of label? I am trying to train a classifier for medical records as a binary classifier. Any suggestions?
Thanks

**Comment by vijay-srivas** [comment]
Hi @oakkas 

Please refer to this link- https://appliedmachinelearning.blog/2019/03/04/state-of-the-art-text-classification-using-bert-model-predict-the-happiness-hackerearth-challenge/

Its explained in a very simple and easy to understand way. Hope it helps. Let me know if its still unclear.

**Comment by jhyuklee** [comment]
Hi @vijay-srivas,

I'm pretty sure that it's a class imbalance problem. You can either undersample your negative samples to make the training dataset class ratio to 1:1 or make more positive samples (oversample) to do the same thing.
Other methodologies like SMOTE might help, but they are out of scope for this work.

Thanks.
Jinhyuk

**Comment by v-loves-avocados** [comment]
Hi @vijay-srivas ,

I am wondering if your problem is solved. I had the same problem. The two labels (0 and 1) in my training and testing both have 1:1 ratio.

Thank you.

**Comment by mohanpradhan87** [comment]
hi. I too am facing the same issue as mentioned above. all the class predictions for run_re.py give negative class (0) with and without data balance. Also i did not find any such issue previously raised. Please reopen this issue.
